client/app.js

```diff
@@ -1,17 +1,112 @@
-const homePage = {template: `
-  <div>home</div>
-`};
-const bookPage = {template: `
-  <div>book {{ $route.params.id }}</div>
-`};
+import { getBooks, getBook, setChapterRead, setChapterUnread, createBook } from './api.js';
+
+const homePage = {
+  template: `
+    <div>
+      <router-link to="/add">Add a book</router-link>
+      <ul v-if="books">
+        <li v-for="book in books" :key="book.id">
+          <router-link :to="'/book/' + book.id">
+            {{ book.title }}
+          </router-link>
+        </li>
+      </ul>
+      <ul v-else>
+        <li>...</li>
+      </ul>
+    </div>
+  `,
+  data: function () {
+    return {
+      books: undefined
+    };
+  },
+  created: function () {
+    getBooks()
+      .then(books => {
+        this.books = books;
+      });
+  },
+};
+const bookPage = {
+  template: `
+    <div>
+      <router-link to="/">Home</router-link>
+      <ul v-if="chapters">
+        <li v-for="chapter in chapters" :key="chapter.id">
+          <my-checkbox
+            :read="!!chapter.read_at"
+            :disabled="loading[chapter.id]"
+            @toggle="toggle(chapter)"
+          ></my-checkbox>
+          {{ chapter.title }} loading: {{ loading[chapter.id] }}
+        </li>
+      </ul>
+      <ul v-else>
+        <li>...</li>
+      </ul>
+    </div>
+  `,
+  data: function () {
+    return {
+      chapters: undefined,
+      loading: undefined,
+    };
+  },
+  created: function () {
+    this.load();
+  },
+  methods: {
+    load: function () {
+    const bookId = this.$route.params.id;
+      getBook(bookId)
+        .then(book => {
+          this.chapters = book.chapters;
+          this.loading = Object.fromEntries(
```

Comment:
Initially this didn't work because I had

```
this.loading = {};
for (let chapter of this.chapters) {
  this.loading[chapter.id] = false;
}
```

but Vue can't detect property addition https://vuejs.org/v2/guide/reactivity.html#For-Objects

This works, though:

```
const loading = {};
for (let chapter of this.chapters) {
  loading[chapter.id] = false;
}
this.loading = loading;
```